subsys/bluetooth/controller/ll_sw/nordic/lll/lll_central_iso.c

```diff
@@ -1237,3 +1133,147 @@ static void isr_done(void *param)
 
 	lll_isr_cleanup(param);
 }
+
+static void payload_count_flush(struct lll_conn_iso_stream *cis_lll)
+{
+	if (cis_lll->tx.bn) {
+		uint64_t payload_count;
+		uint8_t u;
+
+		payload_count = cis_lll->tx.payload_count + cis_lll->tx.bn_curr - 1U;
+		u = cis_lll->nse - ((cis_lll->nse / cis_lll->tx.bn) *
+				    (cis_lll->tx.bn - 1U -
+				     (payload_count % cis_lll->tx.bn)));
+		if (((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr +
+		     cis_lll->tx.ft + u) <
+		    (cis_lll->event_count + se_curr + 2U)) {
+			uint8_t sn_update = cis_lll->tx.bn + 1U - cis_lll->tx.bn_curr;
+
+			/* sn and nesn are 1-bit, only Least Significant bit is needed */
+			cis_lll->sn += sn_update;
+			cis_lll->tx.payload_count += cis_lll->tx.bn;
+			cis_lll->tx.bn_curr = 1U;
+		}
+	}
+
+	if (cis_lll->rx.bn) {
+		uint64_t payload_count;
+		uint8_t u;
+
+		payload_count = cis_lll->rx.payload_count + cis_lll->rx.bn_curr - 1U;
+		u = cis_lll->nse - ((cis_lll->nse / cis_lll->rx.bn) *
+				    (cis_lll->rx.bn - 1U -
+				     (payload_count % cis_lll->rx.bn)));
+		if (((cis_lll->rx.payload_count / cis_lll->rx.bn) + cis_lll->rx.bn_curr +
+		     cis_lll->rx.ft + u) <
+		    (cis_lll->event_count + se_curr + 2U)) {
+			uint8_t nesn_update = cis_lll->rx.bn + 1U - cis_lll->rx.bn_curr;
+
+			/* sn and nesn are 1-bit, only Least Significant bit is needed */
+			cis_lll->nesn += nesn_update;
+			cis_lll->rx.payload_count += cis_lll->rx.bn;
+			cis_lll->rx.bn_curr = 1U;
+		}
+	}
+}
+
+static void payload_count_flush_or_inc_on_close(struct lll_conn_iso_stream *cis_lll)
+{
+	if (cis_lll->tx.bn) {
+		uint64_t payload_count;
+		uint8_t u;
+
+		payload_count = cis_lll->tx.payload_count + cis_lll->tx.bn_curr - 1U;
+		u = cis_lll->nse - ((cis_lll->nse / cis_lll->tx.bn) *
+				    (cis_lll->tx.bn - 1U -
+				     (payload_count % cis_lll->tx.bn)));
+		if ((((cis_lll->tx.payload_count / cis_lll->tx.bn) + cis_lll->tx.bn_curr +
+		      cis_lll->tx.ft + u) <=
+		     (cis_lll->event_count + cis_lll->nse + 2U)) ||
```

Comment:
The comparison between `cis_lll->tx.payload_count / cis_lll->tx.bn` and `event_count` is here yet again different. 

Can you elaborate? 

It think these functions could benefit from some comments